_posts/2024/2024-11-12-Cobbler-v3.3.6-Beginners-Guide.md

```diff
@@ -0,0 +1,560 @@
+reduce network exposure with **firewalld** and secure service/daemon settings
+
+#### Firewalld
+
+Firewalld is the default firewall installed on Fedora and most Red Hat distros, verify that it is enabled and running:
+
+```shell
+systemctl enable --now firewalld
+```
+
+As described above, this Cobbler server has 2 ethernet NICs: **enp0s3** & **enp0s8**
+
+- **enp0s3** being the *upstream* interface which can access the internet
+
+- **enp0s8** being the *downstream* interface which will server the PXE clients
+
+Create a new **firewalld** ***zone*** which **enp0s8** will operate in:
+
+```shell
+firewall-cmd --permanent --new-zone=cobbler
+firewall-cmd --permanent --zone=cobbler --set-target=DROP
+firewall-cmd --permanent --zone=cobbler --add-service=ssh
+firewall-cmd --permanent --zone=cobbler --add-service=tftp
+firewall-cmd --permanent --zone=cobbler --add-service=http
+firewall-cmd --permanent --zone=cobbler --add-service=https
+firewall-cmd --reload
+```
+
+> Linux ISC DHCP server opens a raw UDP socket with the Linux kernel, bypassing the firewalld rules, so it is not necessary to allow the service/port. Be sure the resulting `/etc/dhcp/dhcpd.conf` file is only configured for the desired interface(s) for hosting DHCP.
+> 
+> - [serverfault](https://serverfault.com/questions/191390/iptables-and-dhcp-questions/) & [stackexchange](https://unix.stackexchange.com/questions/447440/ufw-iptables-not-blocking-dhcp-udp-port-67) sources.
+
+Reconfigure **enp0s8** to operate in the **new firewalld zone**
+
+```shell
+nmcli con mod enp0s8 connection.zone cobbler
+```
+
+> As defined above, this zone will NOT permit inbound ICMP traffic like ping
+
+For simplicity, **enp0s3** can be left in the **default firewalld zone** (verify the current active zones with `firewall-cmd --get-active-zones`)
+
+#### Service confinement
+
+On the topic of network security, we should also restrict our Cobbler services to only listen on the PXE server interface.
+
+- configure HTTP to only listen on the PXE server interface:
+
+```shell
+sed -i 's,^Listen.*,Listen 10.0.0.10:80,' /etc/httpd/conf/httpd.conf
+```
+
+ - Create a local systemd service for tftpd, modifiedying the original config to only listen on a specific IP address:
+ 
+```shell
+systemctl stop tftp.service
+systemctl stop tftp.socket
+
+cp /lib/systemd/system/tftp.service /etc/systemd/system/.
+sed 's,ListenDatagram=69,ListenDatagram=10.0.0.10:69,' /lib/systemd/system/tftp.socket | tee /etc/systemd/system/tftp.socket
+```
+
+- now, reload the systemd configuration and restart services
+
+```shell
+systemctl daemon-reload
+systemctl restart tftp httpd
+```
+
+> Feel free to comment with additional security hardening measures you would employ.
+
+## Tips & Troubleshooting
+
+1. Cobbler's [official docs](https://cobbler.readthedocs.io/en/v3.3.6/installation-guide.html#id2) advise that installing and running Cobbler in a virtual environment is not possible, but I have had no issues related to virtualization.
```

Comment:
The issue here are systems dependencies like Fedoras DNF python bindings. As we don't want to assume the users usage profile, we generally recommend against this. Also from a security standpoint, it is much more likely that RPM-based Python packages get proper bugfixes than a venv that is manually maintained by a server admin.

Reply:
I guess by "DNF bindings" or "RPM-based" you mean non obtained through "pip" but I didn't think a virtual server would be any different from a physical server for that purpose. I appreciate your response regardless, I will look into this a little more and leave my footnote as-is in the article for now.

Reply:
No I mean the literal Python DNF bindings. They are only installed once and are only available via system packages, not via pypi:

- https://dnf.readthedocs.io/en/latest/py-modindex.html
- https://pypi.org/project/dnf/

Reply:
Same mechanism is btw also valid for stuff like `aptsources`.

Reply:
But since you are generally right, feel free to leave the footnote. I resolve the conversation in the meantime.